File: VB/WindowsApplication3/MainView.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.Mvvm
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports DXSample.ViewModels


Namespace DXSample
    Partial Public Class MainView
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()
            mvvmContext.RegisterService(DevExpress.Utils.MVVM.Services.LayoutSerializationService.Create(Me, True))
            Dim service As IDocumentManagerService = DirectCast(DevExpress.Utils.MVVM.Services.DocumentManagerService.Create(dockManager), IDocumentManagerService)
            mvvmContext.RegisterService(service)
            Dim fluentAPI = mvvmContext.OfType(Of DocumentsViewModel)()
            fluentAPI.WithEvent(Me, "Load").EventToCommand(Sub(x) x.OnLoaded())
            fluentAPI.WithEvent(Of FormClosingEventArgs)(Me, "FormClosing").EventToCommand(Sub(x) x.OnClosing(Nothing), New Func(Of CancelEventArgs, Object)(Function(args) args))
            fluentAPI.BindCommand(showCustomersBtn, Sub(x, m) x.Show(m), Function(x) x.Modules(0))
            fluentAPI.BindCommand(showOrdersBtn, Sub(x, m) x.Show(m), Function(x) x.Modules(1))
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication3/MainView.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Windows.Forms;
using DevExpress.Mvvm;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using DXSample.ViewModels;


namespace DXSample {
    public partial class MainView : XtraForm {
        public MainView() {
            InitializeComponent();
            mvvmContext.RegisterService(DevExpress.Utils.MVVM.Services.LayoutSerializationService.Create(this, true));
            IDocumentManagerService service = (IDocumentManagerService)DevExpress.Utils.MVVM.Services.DocumentManagerService.Create(dockManager);
            mvvmContext.RegisterService(service);
            var fluentAPI = mvvmContext.OfType<DocumentsViewModel>();
            fluentAPI.WithEvent(this, "Load")
               .EventToCommand(x => x.OnLoaded());
            fluentAPI.WithEvent<FormClosingEventArgs>(this, "FormClosing")
                .EventToCommand(x => x.OnClosing(null), new Func<CancelEventArgs, object>((args) => args));
            fluentAPI.BindCommand(showCustomersBtn, (x, m) => x.Show(m), x => x.Modules[0]);
            fluentAPI.BindCommand(showOrdersBtn, (x, m) => x.Show(m), x => x.Modules[1]);
        }
    }
}

